ProcessLib/SmallDeformationWithLIE/BoundaryCondition/GenericNaturalBoundaryCondition.h

```diff
@@ -0,0 +1,83 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef PROCESSLIB_SMALLDEFORMATION_WITH_LIE_GENERICNATURALBOUNDARYCONDITION_H_
+#define PROCESSLIB_SMALLDEFORMATION_WITH_LIE_GENERICNATURALBOUNDARYCONDITION_H_
+
+#include "MeshLib/MeshSubset.h"
+#include "ProcessLib/BoundaryCondition/BoundaryCondition.h"
+#include "ProcessLib/SmallDeformationWithLIE/Common/FractureProperty.h"
+
+namespace ProcessLib
+{
+class GenericNaturalBoundaryConditionLocalAssemblerInterface;
+
+namespace SmallDeformationWithLIE
+{
+
+template <typename BoundaryConditionData,
+          template <typename, typename, unsigned>
+          class LocalAssemblerImplementation>
+class GenericNaturalBoundaryCondition final : public BoundaryCondition
+{
+public:
+    /// Create a boundary condition process from given config,
+    /// DOF-table, and a mesh subset for a given variable and its component.
+    /// A local DOF-table, a subset of the given one, is constructed.
+    template <typename Data>
+    GenericNaturalBoundaryCondition(
+        typename std::enable_if<
+            std::is_same<typename std::decay<BoundaryConditionData>::type,
+                         typename std::decay<Data>::type>::value,
+            bool>::type is_axially_symmetric,
+        unsigned const integration_order,
+        NumLib::LocalToGlobalIndexMap const& dof_table_bulk,
+        int const variable_id, int const component_id,
+        unsigned const global_dim,
+        std::vector<MeshLib::Element*>&& elements, Data&& data,
+        FractureProperty const& fracture_prop);
```

Comment:
- no, `data` is actually used to pass `parameters`. 
- `template <typename... Data>` cannot simply solve the issue, because this GenericNaturalBoundaryCondition also needs to pass `variable_id` (which is not in the original one) to local assemblers.